library/tasks/TagGeneExon.wdl

```diff
@@ -1,11 +1,35 @@
 task TagGeneExon {
-  # this task requires a gtf file with no internal comments
-  # each record must have a gene_name and transcript_name in addition to a gene_id and transcript_id
-  # the annotation file must be in gtf format (terminal semicolon) not gff format.
-  # the gtf must not have any terminal white space at the end of any line
   File annotations_gtf
   File bam_input
-  Int estimated_required_disk = ceil((size(bam_input, "G") + size(annotations_gtf, "G")) * 2)
+
+  # runtime optional arguments
+  String? opt_docker
+  Int? opt_memory_gb
+  Int? opt_cpu
+  Int? opt_disk
+  Int? opt_preemptible
+
+  # runtime values
+  String docker = select_first([opt_docker, "quay.io/humancellatlas/secondary-analysis-dropseqtools:v0.2.2-1.12"])
+  Int machine_mem_mb = select_first([opt_memory_gb, 7]) * 1000
+  Int cpu = select_first([opt_cpu, 1])
+  # multiply input size by 2
```

Comment:
Probably remove this comment, not informative. (thanks for not deleting my uninformative comments, but feel free :D)